recipes/bx/all/conanfile.py

```diff
@@ -1,81 +1,252 @@
-    options = {
-        "fPIC": [True, False],
-    }
-    default_options = {
-        "fPIC": True,
-    }
+    description = "Base library providing utility functions and macros."
+    topics = ("general", "utility")
+    settings = "os", "compiler", "arch", "build_type"
+    options = {"fPIC": [True, False], "tools": [True, False]}
+    default_options = {"fPIC": True, "tools": False}
 
     @property
-    def _source_subfolder(self):
-        return "source_subfolder"
+    def _bx_folder(self):
+        return "bx"
+
+    @property
+    def _compiler_required(self):
+        return {
+            "gcc": "8",
+            "clang": "3.3",
+            "apple-clang": "5",
+        }
 
     @property
-    def _build_subfolder(self):
-        return "build_subfolder"
+    def _settings_build(self):
+        return getattr(self, "settings_build", self.settings)
 
     def config_options(self):
-        if self.settings.os == 'Windows':
+        if self.settings.os == "Windows":
             del self.options.fPIC
 
     def configure(self):
-        minimal_cpp_standard = "14"
-        if self.settings.compiler.cppstd:
-            tools.check_min_cppstd(self, minimal_cpp_standard)
-        minimal_version = {
-            "gcc": "5",
-            "clang": "3.4",
-            "apple-clang": "10",
-            "Visual Studio": "15"
-        }
-        compiler = str(self.settings.compiler)
-        if compiler not in minimal_version:
-            self.output.warn(
-                "%s recipe lacks information about the %s compiler standard version support" % (self.name, compiler))
-            self.output.warn(
-                "%s requires a compiler that supports at least C++%s" % (self.name, minimal_cpp_standard))
-            return
-        version = tools.Version(self.settings.compiler.version)
-        if version < minimal_version[compiler]:
-            raise ConanInvalidConfiguration("%s requires a compiler that supports at least C++%s" % (self.name, minimal_cpp_standard))
+        if self.settings.os == "Windows":
+            self.libExt = "*.lib"
+            self.binExt = "*.exe"
+            self.packageLibPrefix = ""
+            self.binFolder = "windows"
+        elif self.settings.os in ["Linux", "FreeBSD"]:
+            self.libExt = "*.a"
+            self.binExt = ""
+            self.packageLibPrefix = "lib"
+            self.binFolder = "linux"
+        elif self.settings.os == "Macos":
+            self.libExt = "*.a"
+            self.binExt = ""
+            self.packageLibPrefix = "lib"
+            self.binFolder = "darwin"
+        
+        self.genieExtra = ""
+        self.projs = ["bx"]
+        if self.options.tools:
+            self.projs.extend(["bin2c", "lemon"])
+        if is_msvc(self):
+            # TODO Gotta support old "Visual Studio" compiler until 2.0, then we can remove it
+            if self.settings.compiler == "Visual Studio":
+                if self.settings.compiler.runtime in ["MD", "MDd"]:
```

Comment:
There a helper for static runtime 😉 that does both compiler in 1.x and only the new one in 2.0 its better then this if

Reply:
I'll look for that, didn't know it was a thing!